src/hotspot/share/runtime/globals.hpp

```diff
@@ -1097,6 +1097,7 @@ const int ObjectAlignmentInBytes = 8;
   develop(int, FastAllocateSizeLimit, 128*K,                                \
           /* Note:  This value is zero mod 1<<13 for a cheap sparc set. */  \
           "Inline allocations larger than this in doublewords must go slow")\
+          range(0, (1 << (BitsPerInt - LogBytesPerLong - 1)) - 1)           \
```

Comment:
It would be good to add a comment as to why this specific upper limit is necessary.

Reply:
Thanks, done!